contracts/PriceOracle/PriceOracle.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import {Unauthorized} from "../common/Errors.sol";
+import {System} from "../common/System/System.sol";
+import {HydraChainConnector} from "../HydraChain/HydraChainConnector.sol";
+import {APRCalculatorConnector} from "../APRCalculator/APRCalculatorConnector.sol";
+import {IPriceOracle, PriceForValidator} from "./IPriceOracle.sol";
+
+/**
+ * @title PriceOracle
+ * @dev This contract will be responsible for the price updates.
+ * Active validators will be able to vote and agree on the price.
+ */
+contract PriceOracle is IPriceOracle, System, Initializable, HydraChainConnector, APRCalculatorConnector {
+    mapping(uint256 => PriceForValidator[]) public priceVotesForDay;
+    mapping(address => uint256) public validatorLastVotedDay;
+    mapping(uint256 => uint256) public pricePerDay;
+
+    // _______________ Initializer _______________
+
+    function initialize(address _hydraChainAddr, address _aprCalculatorAddr) external initializer onlySystemCall {
+        __HydraChainConnector_init(_hydraChainAddr);
+        __APRCalculatorConnector_init(_aprCalculatorAddr);
+    }
+
+    // _______________ Modifers _______________
+
+    modifier onlyActiveValidator() {
+        if (hydraChainContract.isValidatorActive(msg.sender) == false) {
```

Comment:
isValidatorActive must be considered based on the realtime ValidatorsData module